src/napari/layers/shapes/_shapes_key_bindings.py

```diff
@@ -158,11 +159,35 @@ def paste_shape(layer: Shapes) -> None:
         layer._paste_data()
 
 
-@register_shapes_action(trans._('Select all shapes in the current view slice'))
+@register_shapes_action(
+    trans._('Select/Deselect all shapes in the current view slice')
+)
 def select_all_shapes(layer: Shapes) -> None:
-    """Select all shapes in the current view slice."""
+    """Select/Deselect all shapes in the current view slice."""
     if layer._mode in (Mode.DIRECT, Mode.SELECT):
-        layer.selected_data = set(np.nonzero(layer._data_view._displayed)[0])
+        new_selected = set(np.nonzero(layer._data_view._displayed)[0])
+
+        if new_selected & layer.selected_data == new_selected:
+            # If all visible shapes are already selected, deselect them
+            layer.selected_data = layer.selected_data - new_selected
+            show_info(
+                trans._(
+                    'Deselected all shapes in this slice, use Shift-A to deselect all shapes on the layer. ({n_total} selected)',
```

Comment:
```suggestion
                    'Deselected all shapes in this slice.',
```
There is no shift-A binding for Shapes, so the rest of the info is spurious.
In fact, could remove the `show_info` entirely? since it is kinda obvious when it happens and there is no other functionality at the moment.